### PR TITLE
Fixed WebAPI - now it works with Simple.OData.Client

### PR DESCRIPTION
**Pull request by benys**
* Fixed ManufacturerApiService - invalid picture service (`PictureService` instead of `IPictureService`)
* Fixed EdmModel to work with Simple.OData.Client - metadata of EDM model was invalid for actions. 
For example CreateProductCategory - previous valid with edm model :
```javascript
{
    'productCategory': {
      'CategoryId': 'id',
      'IsFeaturedProduct': false
    }
}
```
Now I changed Edm model as controller expect:
```javascript
{
  'CategoryId': 'id',
  'IsFeaturedProduct': false
}
```
* Aligned price tag to right
* Added exception if somebody try update stack with warehouseid to product with single mode warehouse.

### CODE REVIEW

File: Grand.Api/Infrastructure/DependencyEdmModel.cs [hunk 1]
```diff
@@ -87,59 +89,87 @@ protected void RegisterProduct(ODataConventionModelBuilder builder)
...
+            deletePicture.Parameter<string>(nameof(ProductPictureDto.PictureId)).Required();
             deletePicture.Returns<bool>();
             #endregion
 
             #region Product specification
             ActionConfiguration createSpecification = product.Action("CreateProductSpecification");
-            createSpecification.Parameter<ProductSpecificationAttributeDto>("productSpecification");
+            createSpecification.Parameter<string>(nameof(ProductSpecificationAttributeDto.Id)).Required();
+            createSpecification.Parameter<int>(nameof(ProductSpecificationAttributeDto.DisplayOrder));
+            createSpecification.Parameter<string>(nameof(ProductSpecificationAttributeDto.CustomValue)).Required();
```
Comment (PatrykPorabik) [inline Grand.Api/Infrastructure/DependencyEdmModel.cs]: In my opinion it shouldn't be required, because it depends on the Attribute Type.

Comment (benys) [reply Grand.Api/Infrastructure/DependencyEdmModel.cs]: You suggest to remove all required modifier or just this one?
Maybe I remove all except  "primarykey" like PicutreId, ManufacturerId etc?

Comment (PatrykPorabik) [reply Grand.Api/Infrastructure/DependencyEdmModel.cs]: Just this one, because the custom values field is not always required.

**Comment by KrzysztofPajak** [comment]
Thank you for contributing!

**Pull request merged**